examples/loss_comparison_2.py

```diff
@@ -183,37 +173,50 @@ def run_experiment(max_iter, max_iter_ref=1000, device=None, seed=None,
             'style': dict(color='tab:red', marker='*', linestyle='--')
+        },
+        # reference cost, use all_n_layers to override the computations
+        'reference': {
+            'label': 'Analysis FISTA',
+            'network': IstaAnalysis,
+            'extra_args': dict(momentum='fista'),
+            'style': dict(color='tab:red', marker='*', linestyle='--'),
+            'all_n_layers': [max_iter_ref]
         }
     }
 
+    for i, learn_prox in enumerate(['none', 'global', 'per-layer']):
+        for n_inner_layer, marker in [(10, '*'), (50, 's'), (100, 'h'),
+                                      (300, 'o'), (500, '>')]:
+            methods[f'lpgd_lista_{learn_prox}_{n_inner_layer}'] = {
+                'label': f'LPGD - LISTA[{learn_prox}-{n_inner_layer}]',
+                'network': ListaTV,
+                'extra_args': dict(n_inner_layers=n_inner_layer,
+                                   learn_prox=learn_prox,
+                                   **learning_parameters),
+                'style': dict(color=f'C{i}', marker=marker, linestyle='-')
+            }
+
     # launch all experiments
     print("=" * 80)
     t0 = time.time()
+    results = Parallel(n_jobs=n_jobs)(
+        delayed(run_one)(x_train, x_test, A, D, L, lbda=lbda, key=k,
+                         network=m['network'], extra_args=m['extra_args'],
+                         all_n_layers=m.get('all_n_layers', all_n_layers),
+                         device=device, meta=meta_pb)
+        for k, m in methods.items()
+    )
+
+    # concatenate all results as a big list. Also update style and label
+    # here to avoid recomputing the results when changing the style only.
     log = []
-    for m in methods.values():
-        print(f"[main script] running {m['label']}")
-        print("-" * 80)
-
-        meta = meta_pb.copy()
-        meta['label'] = m['label']
-        meta['style'] = m['style']
-        results = run_one(x_train, x_test, A, D, L, lbda=lbda,
-                          network=m['network'], extra_args=m['extra_args'],
-                          all_n_layers=all_n_layers, device=device,
-                          meta=meta)
-        log.extend(results)
-
-    # Compute reference cost
-    m = methods['fista_analysis']
-    meta = meta_pb.copy()
-    meta['label'] = 'reference'
-    meta['style'] = None
-    results = run_one(x_train, x_test, A, D, L, lbda=lbda,
-                      network=m['network'], extra_args=m['extra_args'],
-                      all_n_layers=[max_iter_ref], device=device,
-                      meta=meta)
-    log.extend(results[-1:])
+    for records in results:
+        for rec in records:
+            k = rec['key']
+            rec.update(style=methods[k]['style'], label=methods[k]['label'])
+            log.append(rec)
 
+    # Save the computations in a pickle file
     df = pd.DataFrame(log)
     tag = timestamp.strftime('%Y-%m-%d_%Hh%M')
     df.to_pickle(OUTPUT_DIR / f'{SCRIPT_NAME}_{tag}.pkl')
```

Comment:
Cool cool. Store also the `__file__` as emegency back-up?

Reply:
Store the output yes so you can play with `--plot` without breaking everything.